include/seqan3/search/dream_index/interleaved_bloom_filter.hpp

```diff
@@ -603,6 +603,95 @@ class interleaved_bloom_filter<data_layout_mode>::membership_agent::binning_bitv
     friend class membership_agent;
     using sdsl::bit_vector::resize;
     using sdsl::bit_vector::set_int;
+    template <std::integral value_t>
+    friend class counting_vector;
+};
+
+/*!\brief A data structure that behaves like a std::vector and can be used to consolidate the results of multiple calls
+ *        to seqan3::interleaved_bloom_filter::membership_agent::bulk_contains.
+ * \tparam value_t The type of the count. Must model std::integral.
+ *
+ * \details
+ *
+ * When using the seqan3::interleaved_bloom_filter::membership_agent::bulk_contains operation, a common use case is to
+ * add up, for example, the results for all k-mers in a query. This yields, for each bin, the number of k-mers of a
+ * query that are in the respective bin. Such information can be used to apply further filtering or abundance estimation
+ * based on the k-mer counts.
+ *
+ * The seqan3::counting_vector offers an easy way to add up the individual
+ * seqan3::interleaved_bloom_filter::membership_agent::binning_bitvector by offering an `+=` operator.
+ *
+ * The `value_t` template parameter should be chosen in a way that no overflow occurs if all calls to `bulk_contains`
+ * return a hit for a specific bin. For example, `uint8_t` will suffice when processing short Illumina reads, whereas
+ * long reads will require at least `uint32_t`.
+ *
+ * ### Example
+ *
+ * \include test/snippet/search/dream_index/counting_vector.cpp
+ */
+template <std::integral value_t>
+class counting_vector : public std::vector<value_t>
+{
+private:
+    //!\brief The base type.
+    using base_t = std::vector<value_t>;
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    counting_vector() = default; //!< Defaulted.
+    counting_vector(counting_vector const &) = default; //!< Defaulted.
+    counting_vector & operator=(counting_vector const &) = default; //!< Defaulted.
+    counting_vector(counting_vector &&) = default; //!< Defaulted.
+    counting_vector & operator=(counting_vector &&) = default; //!< Defaulted.
+    ~counting_vector() = default; //!< Defaulted.
+
+    using base_t::base_t;
+    //!\}
+
+    /*!\brief Bin-wise adds the bits of a seqan3::interleaved_bloom_filter::membership_agent::binning_bitvector.
+     * \tparam rhs_t The type of the right-hand side.
+     *         Must be seqan3::interleaved_bloom_filter::membership_agent::binning_bitvector.
```

Comment:
This does not correspond to your requires? Change either fo them

Reply:
I cannot require the binning_bitvector because I cannot deduce the a template argument (data_layout)  of the right-hand side of a template member function.
But I can require that the member function is of a certain base class.

What I actually want is what I wrote, what I can do is what I wrote and we do not want to use sdsl in public.